File: Harish Kumar/Array & String Miscellaneous/Maximize Distance to Closest Person/Maximize Distance to Closest Person.java
class Solution {
    public int maxDistToClosest(int[] seats) {
        
        int ans = 0;
        int val = 0;
        
        //case 1 -> sit in the mid of two people
        
        for(int i = 0; i< seats.length;i++)
        {
            if(seats[i] == 1)
            {
                val = 0;
            }
            else
            {
                val++;
                ans = Math.max(ans,(val+1)/2);
            }
        }
        
        //case 2 -> sit in the far left, from first pesron
        
        for(int i = 0; i< seats.length;i++)
        {
            if(seats[i]==1)
            {
                ans = Math.max(ans,i);
                break;
            }
        }
        
        //case 3 --> sit in the far right , from first person
        
        for(int i = seats.length-1; i>= 0;i--)
        {
            if(seats[i] == 1)
            {
                ans = Math.max(ans,seats.length - 1 - i);
                break;
            }
        }
        return ans;
    }
}

File: Aman/miscellaneous/Maximize Distance to Closest Person.cpp
class Solution {
public:
    int maxDistToClosest(vector<int>& seats) {
        int n=seats.size(),j=0,fl=0,res=0;
        for(int i=0;i<n;i++)
        {
            if(seats[i]==1 && fl==0)
            {
                res=max(res,i);
                fl=1;
                j=i;
            }
            else if(seats[i]==1 && fl==1)
            {
                res=max(res,(i-j)/2);
                j=i;
            }
            if(i==n-1 && seats[i]==0)
              res=max(res,i-j);
        }
        return res;
    }
};
